src/structlog/stdlib.py

```diff
@@ -461,7 +461,7 @@ def format(self, record):
             # would transform our dict into a str.
             ed = record.msg.copy()
         except AttributeError:
-            logger = None
+            logger = logging.getLogger(record.name)
```

Comment:
Is there a better way to achieve this? It looks expensive to call getLogger on each logging call.

Reply:
I've got three ideas in my mind:
1. We could make it so that users would have to manually set logger to `ProcessorFormatter` and then make `ProcessorFormatter` to take value from there.
```python
logger = logging.getLogger()
formatter = ProcessorFormatter(JSONRenderer())

handler = StreamHandler(sys.stdout)
handler.setFormatter(formatter)
handler.formatter.logger = logger
# Or via setter which we would create handler.formatter.set_logger(logger)

logger.addHandler(handler)
```

2. Make users patch `logging.Logger.makeRecord` so that it includes `self` as `_logger`.

3. Create some kind of first use only cache (probably some dict with `key` name of logger and `value` logger itself) which would look up logger only first time, store it into cache and take it from there next time.

Do you have anything better in mind?

Reply:
I'd prefer 1). ProcessorFormatter could take the `logger` as a constructor argument.